lib/http.c

```diff
@@ -1577,6 +1577,9 @@ CURLcode Curl_http_connect(struct Curl_easy *data, bool *done)
     result = add_haproxy_protocol_header(data);
     if(result)
       return result;
+
+    /* do not send the header again after successful try */
+    data->set.haproxyprotocol = FALSE;
```

Comment:
Let me know if it's inappropriate to use this flag to keep track whether we already sent the header or not.

Reply:
It is not appropriate. The `set.*` fields need to remain set to the value that was set by the application so that it remains so for future transfers and if the handle is dup'ed etc.

You probably need to invent a state field/bit inside `data->state.` for this.

Reply:
Ah okay, was wondering about that and try to go with the simplest approach, it didn't came to my mind that the handle can be dup'ed.

Lemme create the state then. Thanks!

Reply:
Done.